pyfr/plugins/kernels/turbulence.mako

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<%pyfr:macro name='turbulence' params='t, u, ploc, src'>
+  fpdtype_t tbc = 5.9604644775390625e-8;
+  fpdtype_t arg;  
+  fpdtype_t xloc2;
+  fpdtype_t sclip;
+  fpdtype_t g;
+  fpdtype_t pos[${ndims}];
+  fpdtype_t tploc[${ndims}];
+  fpdtype_t eps[${ndims}];
+  fpdtype_t delta2[${ndims}];
+  fpdtype_t utilde[${ndims}];
+
+  utilde[0] = 0.0;
+  utilde[1] = 0.0;
+  utilde[2] = 0.0;
+
+  unsigned int oldstate;
+  unsigned int newstate;
+  unsigned int rshift;
+
+  int epscomp;
+  
+  % for i, r in enumerate(rot):
+    tploc[${i}] = ${' + '.join(f'{r[j]}*(ploc[{j}] - {shift[j]})' for j in range(3))};
+  % endfor
+
+  % for i in range(nvmax):
+    pos[0] = ${-ls} + (t-tinit[${i}][0])*${ubar};
+
+    % for j in range(2):
+      % if loop.index == 0:
+        oldstate = state[${i}][0];
+      % else:
+        oldstate = newstate;
+      % endif
+      newstate = (oldstate * 747796405UL) + 2891336453UL;
+      rshift = oldstate >> (28UL);
+      oldstate ^= oldstate >> (4UL + rshift);
+      oldstate *= 277803737UL;
+      oldstate ^= oldstate >> 22UL;
+      % if loop.index == 0:
+        pos[1] = ${ymin} + ${ymax-ymin}*((fpdtype_t)(oldstate >> 8UL) * tbc);
+      % else:
+        pos[2] = ${zmin} + ${zmax-zmin}*((fpdtype_t)(oldstate >> 8UL) * tbc);
+      % endif
+    % endfor
+    epscomp = oldstate;
+
+    arg = 0.0;
+    % for j in range(ndims):
+      delta2[${j}] = (pos[${j}]-tploc[${j}])*(pos[${j}]-tploc[${j}]);
+      arg += ${-0.5/(sigma*sigma*ls*ls)}*delta2[${j}];
```

Comment:
Why are we multiplying each time by the same constant?  Factor it out and propagate through into the polyfit.